toolchain/lowering/lowering_context.cpp

```diff
@@ -165,6 +165,23 @@ auto LoweringContext::BuildType(SemanticsNodeId node_id) -> llvm::Type* {
       return llvm::StructType::create(*llvm_context_, subtypes,
                                       "StructLiteralType");
     }
+    case SemanticsNodeKind::TupleType: {
+      auto refs = semantics_ir_->GetTypeBlock(node.GetAsTupleType());
+      llvm::SmallVector<llvm::Type*> subtypes;
+      subtypes.reserve(refs.size());
+      for (auto ref_id : refs) {
+        if (ref_id == SemanticsTypeId::TypeType) {
+          // TODO: Handle TypeType. Currently using a place holder
+          // but this is just a workaround not an actual solution.
+          subtypes.push_back(llvm::Type::getInt1Ty(*llvm_context_));
```

Comment:
What are your thoughts on having the TypeType logic shared in GetType versus here?

Reply:
So far, I haven't come across any other cases that cause the TypeType to be handled except for the tuple. Hence, I added it here as this is tuple specific currently.

Reply:
Here's a case where it happens for structs: `fn F() { {.b = {.a: i32} }; }`

I think this check should be done in `GetType`.